net/kadnode/files/kadnode.init

```diff
@@ -79,6 +79,11 @@ stop_instance() {
 }
 
 start_service() {
+	# wait for localhost
+	while ! ip -4 a s dev lo | grep '127.0.0.1' &> /dev/null; do
+		sleep 1
+	done
```

Comment:
This wait for locahost to be ready is ugly. Suggestions welcome. :-)

Reply:
Some packages like travelmate from @dibdot  have used ubus wait logic to check for interface's presence:
https://github.com/openwrt/packages/blob/master/net/travelmate/files/travelmate.init#L12

You might maybe try something similar.

Reply:
thanks!
I tried "ubus -t 30 wait_for network.interface network.loopback", but it timed out while the loopback interface was running. Maybe it is a problem because the lo state is UNKNOWN on OpenWrt?

Reply:
it is the same for lan. I guess it waits for the signal, thus wait_for :P